File: src/main/java/com/hileco/cortex/instructions/operations/jumps/EXIT.java
package com.hileco.cortex.instructions.operations.jumps;

import com.hileco.cortex.context.ProcessContext;
import com.hileco.cortex.context.ProgramContext;
import com.hileco.cortex.context.ProgramZone;
import com.hileco.cortex.instructions.operations.Operation;

import java.util.Collections;
import java.util.List;

public class EXIT implements Operation {
    public void execute(ProcessContext process, ProgramContext program) {
        process.getPrograms().clear();
    }

    public List<Integer> getStackTakes() {
        return Collections.emptyList();
    }

    public List<Integer> getStackAdds() {
        return Collections.emptyList();
    }

    public List<ProgramZone> getInstructionModifiers() {
        return Collections.singletonList(ProgramZone.PROGRAM_CONTEXT);
    }

    public String toString() {
        return "EXIT";
    }
}

File: cortex-vm/src/main/kotlin/com/hileco/cortex/vm/instructions/jumps/EXIT.kt
package com.hileco.cortex.vm.instructions.jumps

import com.hileco.cortex.vm.instructions.Instruction
import com.hileco.cortex.vm.instructions.InstructionModifier

class EXIT : Instruction() {
    override val instructionModifiers: List<InstructionModifier>
        get() = listOf(InstructionModifier.PROGRAM_CONTEXT)
}
